### PR TITLE
properly encode X-Status

### PR DESCRIPTION
**Pull request by lekoala**
Follow up of
https://github.com/silverstripe/silverstripe-admin/pull/1640#pullrequestreview-1881274997

## Issues
https://github.com/silverstripe/silverstripe-framework/pull/11105
https://github.com/silverstripe/silverstripe-admin/issues/1639

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [ ] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: src/VersionedGridFieldItemRequest.php [hunk 1]
```diff
@@ -26,6 +26,10 @@
  */
 class VersionedGridFieldItemRequest extends GridFieldDetailForm_ItemRequest
 {
+    /**
+     * @param bool $unlinked
+     * @return ArrayList
+     */
```
Comment (ChloeSartorelli) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
```
This method is overridden, and inherits its PHPDoc. No need to duplicate it here.

File: src/VersionedGridFieldItemRequest.php [hunk 2]
```diff
@@ -42,6 +46,7 @@ public function Breadcrumbs($unlinked = false)
         $this->extend('updateBadge', $badge);
 
         if ($badge) {
+            /** @var ArrayData $lastItem */
```
Comment (ChloeSartorelli) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
```
Not needed - we recently added generic typing to lists to handle this

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Please remove these unneeded comment changes - then I'll be happy to merge.

**Comment by GuySartorelli** [comment]
Just noticed it's also targetting the wrong branch - please target this PR at `2.1` so it can be released as a patch. You may also need to [reset your commits](https://gist.github.com/GuySartorelli/74627843aeb75e12b777b1f44cb22d2f) afterwards.

**Comment by lekoala** [comment]
@GuySartorelli i created another pr targeting the right branch turns out it's much easier :)
https://github.com/silverstripe/silverstripe-versioned/pull/440

**Pull request closed without merging**